File: 프로그래머스/lv1/131128. 숫자 짝꿍/숫자 짝꿍.py
def solution(X, Y):
    answer = ''
    x = [0] * 10
    y = [0] * 10
    for num in X: x[int(num)] += 1
    for num in Y: y[int(num)] += 1
    for i, cnt in enumerate(x):
        if cnt != 0 and y[i] != 0:
            answer += str(i) * min(cnt, y[i])
    answer = sorted(list(answer), reverse = True)
    if not answer: return '-1'
    if answer[0] == '0': return '0'
    return ''.join(answer)

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.js
function solution(X, Y) {
    var answer = '';
    
    let mapX = new Map()
    let mapY = new Map()
    
    for (const num of X) {
        mapX.set(num, mapX.has(num) ? mapX.get(num) + 1 : 1)
    }
    for (const num of Y) {
        mapY.set(num, mapY.has(num) ? mapY.get(num) + 1 : 1)
    }
    
    for (const [k, v] of mapX) {
        if (mapY.get(k) == v) answer += k.repeat(v)
        else answer += k.repeat(Math.min(mapY.get(k), v))
    }
    
    if (answer.length === 0) return '-1'
    if (answer.length === answer.split('').filter((v) => v === '0').length) return '0'
    
    return answer.split('').sort((a, b) => b-a).join('')
    
}